roles/libvirt_manager/tasks/create_cloud_init_iso.yml

```diff
@@ -0,0 +1,72 @@
+---
+- name: Generate cloud-init iso
+  become: false
+  vars:
+    _images_dir: "{{ [ cifmw_libvirt_manager_basedir, 'workload', ] | ansible.builtin.path_join }}"
+  block:
+    - name: "Acquire authorized_keys from hypervisor"
+      ansible.builtin.slurp:
+        src: "{{ ansible_user_dir ~ '/.ssh/authorized_keys' }}"
+      register: _authorized_keys_b64
+
+    - name: Create a list of ssh pub keys from authorized_keys
+      ansible.builtin.set_fact:
+        _ssh_authorized_keys: |-
+          [
+          {% for ssh_key in (_authorized_keys_b64.content | b64decode).split('\n') | select() %}
+            "{{ ssh_key | trim }}",
+          {% endfor %}
+          ]
+
+    - name: "Generate instance random UUID"
+      ansible.builtin.set_fact:
+        _instance_uuid: "{{ 99999999 | random(seed=vm) | to_uuid | lower }}"
+
+    - name: "Define the network config if needed"
+      when: item.value.ip_address is defined
+      ansible.builtin.set_fact:
+        _network_config:
+          network:
+            version: 2
+            renderer: NetworkManager
+            ethernets:
+              match:
+                name: eth0
```

Comment:
I'm not sure if hard-coding eth0 will escalate properly...

Reply:
I agree but i have just translated the [old](https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/libvirt_manager/templates/static_ip.j2) bash script. I never seen a real life example where we need to define this (and by this I mean where we define "ip_address" in the vm layout, so we probably can get rid of it.

Reply:
@cjeanner do you know something more about it?

Reply:
We should have the MAC address available, it would be better to match on that?

Also, each of the "eternets" connection need to be defined under a "connaction id" key - see: https://cloudinit.readthedocs.io/en/latest/reference/network-config-format-v2.html#device-configuration-ids

```
  ethernets:
    <connection_id>:
      match:
        macaddress: 52:54:00:bd:8f:cb
      set-name: eth0
      addresses:
        - ...
```

I think it would be a good idea to use set-name to define an interface name matching the name of network. like `osptrunk`, `ctlplane` etc - matching the name of network as defined in libvirt manager config?

Reply:
As i thought, this is not required anymore. I chose the KISS approach here, leaving to the user the ability to pass a dict based on `network-config` v2.